sylt-typechecker/src/typechecker.rs

```diff
@@ -1018,8 +1018,10 @@ mod op {
 
     pub fn less(a: &Type, b: &Type) -> Type {
         match (a, b) {
-            (Type::Float, Type::Float) => Type::Bool,
-            (Type::Int, Type::Int) => Type::Bool,
+            (Type::Float, Type::Float)
+            | (Type::Int, Type::Int)
+            | (Type::Float, Type::Int)
+            | (Type::Int, Type::Float) => Type::Bool,
```

Comment:
You can also use an or-pattern for fewer lines.

Reply:
You mean this:
https://github.com/rust-lang/rfcs/blob/master/text/2535-or-patterns.md#dont-repeat-yourself
?

I thought I tried that some weeks ago and it was "experimental" or something. I'll look into this tho.

Reply:
Just gonna leave this here...
```
error[E0658]: or-patterns syntax is experimental
    --> sylt-typechecker/src/typechecker.rs:1021:14
     |
1021 |             (Type::Float | Type::Int, Type::Float | Type::Int) => Type::Bool,
     |              ^^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: see issue #54883 <https://github.com/rust-lang/rust/issues/54883> for more information

error[E0658]: or-patterns syntax is experimental
    --> sylt-typechecker/src/typechecker.rs:1021:39
     |
1021 |             (Type::Float | Type::Int, Type::Float | Type::Int) => Type::Bool,
     |                                       ^^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: see issue #54883 <https://github.com/rust-lang/rust/issues/54883> for more information

error: aborting due to 2 previous errors
```